services/src/handlers/workflows.rs

```diff
@@ -71,16 +72,21 @@ async fn get_workflow_metadata<C: Context>(
         .load(&WorkflowId(id))
         .await?;
 
-    let operator = workflow.operator.get_vector().context(error::Operator)?;
-
-    let operator = operator
-        .initialize(&ctx.execution_context()?)
-        .context(error::Operator)?;
+    let execution_context = ctx.execution_context()?;
 
     // TODO: use cache here
-    let result_descriptor = operator.result_descriptor();
-
-    Ok(warp::reply::json(result_descriptor))
+    call_on_typed_operator!(
```

Comment:
do we really need the macro here? I fear that it somehow hides the initialize step and we should not initialize workflows more then needed. I know it i needed at the moment.

Reply:
The macro doesn't cover the initialization, you see it in the function where it is used. So I can't share this fear.

Reply:
it's ok like this